include/deal.II/numerics/fe_field_function.h

```diff
@@ -157,6 +151,11 @@ namespace Functions
    *     ...do something...;
    * @endcode
    *
+   * @note If points are repeated among processes the user should take special
+   * care because FEFieldFunction returning the value on ghost cells might result
+   * in multiple processes evaluating the solution at the same point, with no clear
+   * process responsible for producing output.
+   *
```

Comment:
This should not be a concern of the user. Unless your goal is to change this right afterwards, our goal was to change the implementation of `FEFieldFunction` for distributed grids so that you could query the values anywhere, and it would return it correctly (even if with communications, when you are on artificial nodes)

Reply:
I agree: the point of this PR is that currently the documentation there states:
1. if the point is not on a locally owned cell (i.e. it is on a ghost cell or an artificial cell) then an exception is triggered.
2. we don't want to compute things on ghost cells to avoid multiple processes computing the same thing.

The 1. is false: the exception is triggered only for artifical cells (I've now updated the documentation for all functions), while for 2. I added that note.

Now I removed it, since you've commented about it but I wasn't the one to write that concept in the documentation ;)

Reply:
Just squash and I'll merge. Any improvement in the documentation is a good thing. 

Thanks for doing this!